clients/drcachesim/tracer/output.cpp

```diff
@@ -1347,18 +1347,14 @@ exit_thread_io(void *drcontext)
             BUF_PTR(data->seg_base), dr_get_thread_id(drcontext));
 
         ptr_int_t window = get_local_window(data);
-        if (has_tracing_windows()) {
-            if (is_in_tracing_mode(tracing_mode.load(std::memory_order_acquire)) &&
-                get_local_window(data) == window) {
-                process_and_output_buffer(
-                    drcontext,
-                    /* If this thread already wrote some data, include
-                     * its exit even if we're over a size limit.
-                     */
-                    data->bytes_written > 0);
-            }
-        } else {
-            process_and_output_buffer(drcontext, data->bytes_written > 0);
+        if (!has_tracing_windows() ||
+            (has_tracing_windows() &&
```

Comment:
This can be removed as it is redundant.

Reply:
Done. simplified to `!has_tracing_windows() || is_in_tracing_mode` to make sure we are checking if is in tracing mode when we are window tracing.